### PR TITLE
jellyfin{,-web}: 10.7.7 -> 10.8.0

### PR DESCRIPTION
**Pull request by jojosch**
###### Motivation for this change
Update to latest stable release https://github.com/jellyfin/jellyfin/releases/tag/v10.8.0. 

Release blogpost: https://jellyfin.org/posts/jellyfin-10-8-0/

Now `jellyfin-web` is using npm with a lockfile - so the `update-web.sh` can use that lockfile and we don't need to patch the unsupported dependency url in `postFetch`.

Tested by running a 10.7.7 server and upgrading to 10.8.0. The nixos tests also pass.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by purcell** [comment]
Nice, thanks for working on this. Changes look good to me.

**Review by IvarWithoutBones — APPROVED** [review]
LGTM, thanks!

**Comment by Kranzes** [comment]
I believe that before Jellyfin gets updated in nixpkgs, these PRs need to be handled: https://github.com/NixOS/nixpkgs/pull/149715, https://github.com/NixOS/nixpkgs/pull/108224 and perhaps even https://github.com/NixOS/nixpkgs/pull/151617.

**Comment by justinas** [comment]
> I believe that before Jellyfin gets updated in nixpkgs, these PRs need to be handled: #149715, #108224 and perhaps even #151617.

@Kranzes I do not think `jellyfin-ffmpeg` should block this PR in case 10.8.0 is finalized soon. We have lived long enough without it here, so :)

**Comment by Kranzes** [comment]
Then I guess fixing NVENC transcoding is important enough to block this PR.

**Comment by jojosch** [comment]
Rebased and updated to https://github.com/jellyfin/jellyfin/releases/tag/v10.8.0-beta1

**Comment by jojosch** [comment]
Rebased and updated to https://github.com/jellyfin/jellyfin/releases/tag/v10.8.0-beta2

**Comment by jojosch** [comment]
Rebased and updated to https://github.com/jellyfin/jellyfin/releases/tag/v10.8.0-beta3

**Comment by jojosch** [comment]
Rebased and updated to stable release https://github.com/jellyfin/jellyfin/releases/tag/v10.8.0.

**Review by bachp — APPROVED** [review]
Deployed 10.8.0 and so far everything is working

**Comment by peperunas** [comment]
Lgtm!

**Comment by AkechiShiro** [comment]
What's blocking this merge ? Maybe there needs a second approval ?

File: nixos/tests/jellyfin.nix
Comment (minijackson) [inline nixos/tests/jellyfin.nix:26]: If you want to fix the typechecking, here is a patch you can apply: https://gist.github.com/minijackson/66f7b1f2b6699005e56d5ef8e5247422

If you don't want to modify your PR, no worries, I can make one afterwards.

**Review by minijackson — APPROVED** [review]
Haven't tested it, but looks awesome, thanks a lot :heart:

File: nixos/tests/jellyfin.nix
Comment (jojosch) [reply nixos/tests/jellyfin.nix:26]: Applied your patch to this PR. Thank you.

**Review by minijackson — APPROVED** [review]

**Comment by onny** [comment]
I would like to use the new version on NixOS 22.05 stable channel.

I tried to reference the unstable package in the Jellyfin module but it fails with following error:

```
[...]
unpacking sources
unpacking source archive /nix/store/95dhgdvzqfbqhzs05nzy061skg4dnqi8-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/jay0cxwr28rvdr66s654hz01pj0fccri-disable-warnings.patch
patching file jellyfin.ruleset
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
Executing dotnetConfigureHook
  Determining projects to restore...
/build/source/Jellyfin.Server.Implementations/Jellyfin.Server.Implementations.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Run>
/build/source/Jellyfin.Server.Implementations/Jellyfin.Server.Implementations.csproj : error NU1101: Unable to find package Microsoft.AspNetCore.App.>
  Failed to restore /build/source/Jellyfin.Server.Implementations/Jellyfin.Server.Implementations.csproj (in 39.35 sec).
/build/source/Jellyfin.Networking/Jellyfin.Networking.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.linux-arm64. No pac>
/build/source/Jellyfin.Networking/Jellyfin.Networking.csproj : error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.linux-arm64. No >
  Failed to restore /build/source/Jellyfin.Networking/Jellyfin.Networking.csproj (in 247 ms).
/build/source/Jellyfin.Drawing.Skia/Jellyfin.Drawing.Skia.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.linux-arm64. No>
/build/source/Jellyfin.Drawing.Skia/Jellyfin.Drawing.Skia.csproj : error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.linux-arm64.>
  Failed to restore /build/source/Jellyfin.Drawing.Skia/Jellyfin.Drawing.Skia.csproj (in 9.36 sec).
/build/source/Jellyfin.Data/Jellyfin.Data.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.linux-arm64. No packages exist >
/build/source/Jellyfin.Data/Jellyfin.Data.csproj : error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.linux-arm64. No packages exi>
  Failed to restore /build/source/Jellyfin.Data/Jellyfin.Data.csproj (in 8.97 sec).
/build/source/Jellyfin.Api/Jellyfin.Api.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.linux-arm64. No packages exist wi>
/build/source/Jellyfin.Api/Jellyfin.Api.csproj : error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.linux-arm64. No packages exist>
  Failed to restore /build/source/Jellyfin.Api/Jellyfin.Api.csproj (in 4.66 sec).
/build/source/Emby.Drawing/Emby.Drawing.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.linux-arm64. No packages exist wi>
/build/source/Emby.Drawing/Emby.Drawing.csproj : error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.linux-arm64. No packages exist>
  Failed to restore /build/source/Emby.Drawing/Emby.Drawing.csproj (in 194 ms).
/build/source/Emby.Dlna/Emby.Dlna.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.linux-arm64. No packages exist with thi>
/build/source/Emby.Dlna/Emby.Dlna.csproj : error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.linux-arm64. No packages exist with >
  Failed to restore /build/source/Emby.Dlna/Emby.Dlna.csproj (in 211 ms).
/build/source/DvdLib/DvdLib.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.linux-arm64. No packages exist with this id i>
/build/source/DvdLib/DvdLib.csproj : error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.linux-arm64. No packages exist with this i>
  Failed to restore /build/source/DvdLib/DvdLib.csproj (in 36 ms).
[...]
```

**Comment by jojosch** [comment]
@onny Building on `arm64` should be fixed with #177902

**Comment by AkechiShiro** [comment]
Is there any status where we can check that this update is in NixOS unstable channel ? I still didn't receive this update.

**Comment by onny** [comment]
> @onny Building on `arm64` should be fixed with #177902

Yep, works for me now, thank you

**Comment by minijackson** [comment]
@AkechiShiro this update is in the NixOS unstable channel, you can look it up on the nixpk.gs pull request tracker here: https://nixpk.gs/pr-tracker.html?pr=144082

**Comment by AkechiShiro** [comment]
Thanks @minijackson I had to reboot and update my kernel to receive the update I'm not sure why

**Pull request merged**
